dhall/src/Dhall/Eval.hs

```diff
@@ -1,106 +1,79 @@
-{-# LANGUAGE
-  BangPatterns,
-  CPP,
-  LambdaCase,
-  OverloadedStrings,
-  PatternSynonyms,
-  RankNTypes,
-  ScopedTypeVariables,
-  TupleSections,
-  ViewPatterns
-  #-}
-
-{-# OPTIONS_GHC
-  -O
-  -fno-warn-name-shadowing
-  -fno-warn-unused-matches
-  #-}
-
-{-|
-Eval-apply environment machine with conversion checking and quoting to normal
-forms. Fairly similar to GHCI's STG machine algorithmically, but much simpler,
-with no known call optimization or environment trimming.
-
-Potential optimizations without changing Expr:
-  - In conversion checking, get non-shadowing variables not by linear
-    Env-walking, but by keeping track of Env size, and generating names which
-    are known to be illegal as source-level names (to rule out shadowing).
-  - Use HashMap Text chunks for large let-definitions blocks. "Large" vs "Small"
-    is fairly cheap to determine at evaluation time.
-
-Potential optimizations with changing Expr:
-  - Use Int in Var instead of Integer. Overflow is practically impossible.
-  - Use actual full de Bruijn indices in Var instead of Text counting indices. Then, we'd
-    switch to full de Bruijn levels in Val as well, and use proper constant time non-shadowing
-    name generation.
+{-# LANGUAGE BangPatterns        #-}
+{-# LANGUAGE CPP                 #-}
+{-# LANGUAGE LambdaCase          #-}
+{-# LANGUAGE OverloadedStrings   #-}
+{-# LANGUAGE PatternSynonyms     #-}
+{-# LANGUAGE RankNTypes          #-}
+{-# LANGUAGE ScopedTypeVariables #-}
+{-# LANGUAGE TupleSections       #-}
+{-# LANGUAGE ViewPatterns        #-}
 
+{-# OPTIONS_GHC -O -fno-warn-name-shadowing -fno-warn-unused-matches #-}
+
+{-| Eval-apply environment machine with conversion checking and quoting to
+    normal forms. Fairly similar to GHCI's STG machine algorithmically, but much
+    simpler, with no known call optimization or environment trimming.
+
+    Potential optimizations without changing Expr:
+
+    * In conversion checking, get non-shadowing variables not by linear
+      Env-walking, but by keeping track of Env size, and generating names which
+      are known to be illegal as source-level names (to rule out shadowing).
+
+    * Use HashMap Text chunks for large let-definitions blocks. "Large" vs
+      "Small" is fairly cheap to determine at evaluation time.
+
+    Potential optimizations with changing Expr:
+
+    * Use actual full de Bruijn indices in Var instead of Text counting indices.
+      Then, we'd switch to full de Bruijn levels in Val as well, and use proper
+      constant time non-shadowing name generation.
 -}
 
 module Dhall.Eval (
-    Env(..)
-  , Names(..)
-  , Closure(..)
-  , VChunks(..)
-  , Val(..)
-  , Void
-  , inst
-  , eval
-  , conv
-  , convEmpty
-  , quote
-  , nf
-  , nfEmpty
-  , Dhall.Eval.alphaNormalize
+    judgmentallyEqual
+  , normalize
+  , alphaNormalize
   ) where
 
 #if MIN_VERSION_base(4,8,0)
```

Comment:
We shouldn't need this anymore: `dhall` has a constraint ` base >=4.8.2.0`.

We can then also get rid of the `LANGUAGE CPP` pragma I believe.